Make validation markup more generic

katzmo opened the pull request.

katzmo committed:
JS: Trigger proper clientside validation on error

Use jQuery.validate’s showErrors() function to display error messages
related to form fields. No need to replicate clientside validation
markup. However, a name property is required for an element to call
showErrors() on, forcing us to generate invalid html.

katzmo committed:
JS: render remaining stripe.js errors

When there is a general error or the field is not on this step:
- Add message to list of clientside validation errors when there is a
dedicated container (usually when all errors are displayed in one place)
- If not, show message above the payment fieldset using markup from the
clientside validation settings (errorClass is hardcoded there, so might
not provide the expected markup when overriden in custom error function)

When clientside validation is not used:
- Add hardcoded markup above the form using the Drupal standard message
classes (again, could be differently themed for a particular site and
might need extra styling).

torotil commented on package/src/method-element.js:
```diff
@@ -153,7 +153,7 @@ class MethodElement {
   validate (submitter) {
     $('.mo-dialog-wrapper').addClass('visible')
     if (typeof Drupal.clientsideValidation !== 'undefined') {
-      $('#clientsidevalidation-' + this.form_id + '-errors ul').empty()
```
Is there a chance that clientside validation is enabled for *other* forms on the same page, but isn’t enabled for *this* form?

katzmo replied:
Theoretically, I guess it's possible…?

torotil commented on package/src/method-element.js:
```diff
@@ -215,14 +216,30 @@ class MethodElement {
+    // Without clientside validation render a message above the fieldset.
+    else {
+      const $message = $('<div class="messages error">').text(error.message)
+      $message.addClass('stripe-error').insertBefore(this.$element.closest('form'))
```
This is above the form (as stated in the commit message) not the fieldset (as stated in the comment), right?

katzmo replied:
yes ;)

katzmo committed:
JS: improve check for clientside validation.

katzmo committed:
JS: handle errors on change.

katzmo committed:
JS: fix clientside validation form reset.

We just want to clear the error messages, resetForm() also removes the
form values.
